src/seedu/addressbook/data/person/Person.java

```diff
@@ -14,7 +17,10 @@
     private Phone phone;
     private Email email;
     private Address address;
-
+    private ArrayList<Tagging> tagsHistory = new ArrayList<Tagging>();
```

Comment:
this may not be the right place to maintain the tag history.
you are required to maintain the list of tagging for the address book  session, not the person.